Detectors/TRD/calibration/src/CalibratorVdExB.cxx

```diff
@@ -101,57 +104,104 @@ void CalibratorVdExB::initProcessing()
   if (mInitDone) {
     return;
   }
+
   mFitFunctor.lowerBoundAngleFit = 80 * TMath::DegToRad();
   mFitFunctor.upperBoundAngleFit = 100 * TMath::DegToRad();
-  mFitFunctor.vdPreCorr = 1.546;    // TODO: will be taken from CCDB in the future
-  mFitFunctor.laPreCorr = 0.;       // TODO: will be taken from CCDB in the future
+  mFitFunctor.mAnodePlane = GeometryBase::camHght() / (2.f * 100.f);
   for (int iDet = 0; iDet < MAXCHAMBER; ++iDet) {
     mFitFunctor.profiles[iDet] = std::make_unique<TProfile>(Form("profAngleDiff_%i", iDet), Form("profAngleDiff_%i", iDet), NBINSANGLEDIFF, -MAXIMPACTANGLE, MAXIMPACTANGLE);
   }
+
+  mFitter.SetFCN<FitFunctor>(2, mFitFunctor, mParamsStart);
+  mFitter.Config().ParSettings(ParamIndex::LA).SetLimits(-0.7, 0.7);
+  mFitter.Config().ParSettings(ParamIndex::LA).SetStepSize(.01);
+  mFitter.Config().ParSettings(ParamIndex::VD).SetLimits(0., 3.);
+  mFitter.Config().ParSettings(ParamIndex::VD).SetStepSize(.01);
+  ROOT::Math::MinimizerOptions opt;
+  opt.SetMinimizerType("Minuit2");
+  opt.SetMinimizerAlgorithm("Migrad");
+  opt.SetPrintLevel(0);
+  opt.SetMaxFunctionCalls(1'000);
+  opt.SetTolerance(.001);
+  mFitter.Config().SetMinimizerOptions(opt);
+
   mInitDone = true;
 }
 
+void CalibratorVdExB::retrievePrev(o2::framework::ProcessingContext& pc)
+{
+  static bool doneOnce = false;
+  if (!doneOnce) {
+    doneOnce = true;
+    mFitFunctor.vdPreCorr.fill(1.546);
+    mFitFunctor.laPreCorr.fill(0.);
+    // We either get a pointer to a valid object in the last error or to the default object
+    // which is always present. The first has precedence over the latter.
+    auto dataCalVdriftExB = pc.inputs().get<o2::trd::CalVdriftExB*>("calvdexb");
+    if (!dataCalVdriftExB) {
```

Comment:
In this case you would get a fatal error from the framework, so no need to check this

Reply:
Ok.